corehq/apps/geospatial/management/commands/index_geolocation_case_properties.py

```diff
@@ -38,51 +33,35 @@ def handle(self, *args, **options):
 
 def index_case_docs(domain, query_limit=DEFAULT_QUERY_LIMIT, chunk_size=DEFAULT_CHUNK_SIZE, case_type=None):
     geo_case_property = get_geo_case_property(domain)
-    query = _es_case_query(domain, geo_case_property, case_type)
+    query = case_query_for_missing_geopoint_val(domain, geo_case_property, case_type)
     count = query.count()
     print(f'{count} case(s) to process')
-    batch_count = 1
-    if query_limit:
-        batch_count = math.ceil(count / query_limit)
+    batch_count = get_batch_count(count, query_limit)
     print(f"Cases will be processed in {batch_count} batches")
     for i in range(batch_count):
         print(f'Processing {i+1}/{batch_count}')
-        query = _es_case_query(domain, geo_case_property, case_type, size=query_limit)
-        case_ids = query.get_ids()
-        _index_case_ids(domain, case_ids, chunk_size)
+        process_batch(domain, geo_case_property, case_type, query_limit, chunk_size, with_progress=True)
 
 
-def _index_case_ids(domain, case_ids, chunk_size):
-    for case_id_chunk in chunked(with_progress_bar(case_ids), chunk_size):
-        case_chunk = CommCareCase.objects.get_cases(list(case_id_chunk), domain)
-        case_search_adapter.bulk_index(case_chunk)
-    manager.index_refresh(case_search_adapter.index_name)
+def get_batch_count(doc_count, query_limit):
```

Comment:
nit: I'd suggest getting rid of this function.
1. it adds to what is being discussed in https://github.com/dimagi/commcare-hq/pull/35126/files#r1782643923
2. I am not sure its adding enough value as a function. At places where query_limit is always present you could simply use, `math.ceil(doc_count / query_limit)`

though, its mainly the first point for me.

Reply:
That is a good point. I initially added this as a safeguard for if we ever receive a `query_limit` of 0 in the `index_geolocation_case_properties` command, and simply extended this util function to be used in the task as well. I agree this could be removed entirely and simplified.

I'll do a refactor to remove this util completely.

Reply:
Addressed in [eb45210](https://github.com/dimagi/commcare-hq/pull/35126/commits/eb452105306e4ef612d9e2ca370a39c397d9c6da).